pkg/service/restore/worker.go

```diff
@@ -0,0 +1,887 @@
+				CreateStmt: stmt,
+			})
+		}
+
+		for _, view := range meta.Views {
+			if !restoredTables.Has(view.KeyspaceName + "." + view.BaseTableName) {
+				continue
+			}
+			dummyMeta := gocql.KeyspaceMetadata{
+				Views: map[string]*gocql.ViewMetadata{view.ViewName: view},
+			}
+
+			schema, err := dummyMeta.ToCQL()
+			if err != nil {
+				return errors.Wrapf(err, "get view %s.%s create statement", ks, view.ViewName)
+			}
+
+			// DummyMeta schema consists of create keyspace and create view statements
+			stmt := strings.Split(schema, ";")[1]
+			stmt, err = addIfNotExists(stmt, MaterializedView)
+			if err != nil {
+				return err
+			}
+
+			views = append(views, View{
+				Keyspace:   view.KeyspaceName,
+				View:       view.ViewName,
+				Type:       MaterializedView,
+				BaseTable:  view.BaseTableName,
+				CreateStmt: stmt,
+			})
+		}
+	}
+
+	w.logger.Info(ctx, "Initialized views", "views", views)
+	w.run.Views = views
+
+	return nil
+}
+
+// cleanUploadDir deletes all SSTables from host's upload directory except for those present in excluded.
+func (w *worker) cleanUploadDir(ctx context.Context, host, dir string, excluded []string) error {
+	var toBeDeleted []string
+	s := strset.New(excluded...)
+	opts := &scyllaclient.RcloneListDirOpts{FilesOnly: true}
+
+	err := w.client.RcloneListDirIter(ctx, host, dir, opts, func(item *scyllaclient.RcloneListDirItem) {
+		if !s.Has(item.Name) {
+			toBeDeleted = append(toBeDeleted, item.Name)
+		}
+	})
+	if err != nil {
+		return errors.Wrapf(err, "list dir: %s on host: %s", dir, host)
+	}
+
+	if len(toBeDeleted) > 0 {
+		w.logger.Info(ctx, "Delete files from host's upload directory",
+			"host", host,
+			"upload_dir", dir,
+			"files", toBeDeleted,
+		)
+	}
+
+	for _, f := range toBeDeleted {
+		remotePath := path.Join(dir, f)
+		if err := w.client.RcloneDeleteFile(ctx, host, remotePath); err != nil {
+			return errors.Wrapf(err, "delete file: %s on host: %s", remotePath, host)
+		}
+	}
+	return nil
+}
+
+// Docs: https://docs.scylladb.com/stable/cql/ddl.html#tombstones-gc-options.
+type tombstoneGCMode string
+
+const (
+	modeDisabled  tombstoneGCMode = "disabled"
+	modeTimeout   tombstoneGCMode = "timeout"
+	modeRepair    tombstoneGCMode = "repair"
+	modeImmediate tombstoneGCMode = "immediate"
+)
+
+// AlterTableTombstoneGC alters 'tombstone_gc' mode.
+func (w *worker) AlterTableTombstoneGC(ctx context.Context, keyspace, table string, mode tombstoneGCMode) error {
+	w.logger.Info(ctx, "Alter table's tombstone_gc mode",
+		"keyspace", keyspace,
+		"table", table,
+		"mode", mode,
+	)
+
+	op := func() error {
+		return w.clusterSession.ExecStmt(alterTableTombstoneGCStmt(keyspace, table, mode))
+	}
+
+	notify := func(err error, wait time.Duration) {
+		w.logger.Info(ctx, "Altering table's tombstone_gc mode failed",
+			"keyspace", keyspace,
+			"table", table,
+			"mode", mode,
+			"error", err,
+			"wait", wait,
+		)
+	}
+
+	return alterSchemaRetryWrapper(ctx, op, notify)
+}
+
```

Comment:
`worker.go` is a very big file already.
One of the options to split it into smaller chunks is to divide it by logic.

for example:
- extract `GetTableTombstoneGCMode` and `AlterTableTombstoneGC` and related to another file called `worker_tombstoneGC`

The same applies to other smaller chunks of worker.

Pls let me know what do you thinkg about it.

Reply:
I added the following files:
- `worker_tombstonegc`
- `worker_views`
- `worker_manifest`